docker-compose.yml

```diff
@@ -35,13 +35,43 @@ services:
     #   docker compose build java
     #   docker compose run java
     # Parameters:
-    #   MAVEN: 3.9.6
+    #   MAVEN: 3.9.9
     #   JDK: 11, 17, 21
     image: ${ARCH}/maven:${MAVEN}-eclipse-temurin-${JDK}
-    volumes: &java-volumes
+    volumes:
       - .:/arrow-java:delegated
       - ${DOCKER_VOLUME_PREFIX}maven-cache:/root/.m2:delegated
-    command: &java-command >
+    command:
       /bin/bash -c "
         /arrow-java/ci/scripts/java_build.sh /arrow-java /build &&
         /arrow-java/ci/scripts/java_test.sh /arrow-java /build"
+
+  conda-java-jni-cdata:
```

Comment:
We may want to remove the `-java` part because this repository is only for the Java implementation.